### PR TITLE
feat(snomed.datastore): Adjust page size dynamically for streaming queries

### PR DESCRIPTION
**Pull request by apeteri**
The limit should never exceed the value of `result_window` configured for Elasticsearch.

### CODE REVIEW

**Review by nagyo — APPROVED** [review]
LGTM!

File: commons/com.b2international.index.tests/src/com/b2international/index/StreamTest.java [hunk 1]
```diff
@@ -104,9 +104,11 @@ public void searchAndReturnAllHitsWithStream() throws Exception {
 		List<Data> hits = newArrayListWithExpectedSize(NUM_DOCS);
 		
 		Stopwatch w = Stopwatch.createStarted();
+		
+		final int pageSize = Integer.parseInt((String) getIndexSettings().getOrDefault(IndexClientFactory.RESULT_WINDOW_KEY, "" + IndexClientFactory.DEFAULT_RESULT_WINDOW));
 		final Query<Data> query = Query.select(Data.class)
 			.where(Expressions.matchAll())
-			.limit(1000)
+			.limit(pageSize)
```
Comment (cmark) [inline commons/com.b2international.index.tests/src/com/b2international/index/StreamTest.java]: I think the purpose here was to actually page through the results with the page size of `1000`, now we are going to fetch it in one go, right?

File: commons/com.b2international.index/src/com/b2international/index/es/EsDocumentSearcher.java [hunk 2]
```diff
@@ -194,7 +194,7 @@ public <T> Hits<T> search(Query<T> query) throws IOException {
 		final boolean isLiveStreaming = !Strings.isNullOrEmpty(query.getSearchAfter());
 		if (isLocalStreaming) {
 			checkArgument(!isLiveStreaming, "Cannot use searchAfter when requesting more items (%s) than the configured result window (%s).", limit, resultWindow);
-		} else if (isLiveStreaming) {
+		} if (isLiveStreaming) {
```
Comment (cmark) [inline commons/com.b2international.index/src/com/b2international/index/es/EsDocumentSearcher.java]: This block looks a bit weird.

**Review by cmark — CHANGES_REQUESTED** [review]

**Review by cmark — APPROVED** [review]
🥓

**Pull request merged**